AutomatedTesting/Gem/PythonTests/scripting/Node_HappyPath_DuplicateNode.py

```diff
@@ -34,80 +46,67 @@ def Node_HappyPath_DuplicateNode():
         background = scroll_area.findChild(QtWidgets.QFrame, "Background")
-        title = background.findChild(QtWidgets.QLabel, "Title")
-        text = title.findChild(QtWidgets.QLabel, "Title")
+        title = background.findChild(QtWidgets.QLabel, TITLE_STRING)
+        text = title.findChild(QtWidgets.QLabel, TITLE_STRING)
         return text.text()
 
-    # 1) Open Script Canvas window (Tools > Script Canvas)
-    general.idle_enable(True)
-    general.open_pane("Script Canvas")
-    helper.wait_for_condition(lambda: general.is_pane_visible("Script Canvas"), 5.0)
-
-    # 2) Open a new graph
-    editor_window = pyside_utils.get_editor_main_window()
-    sc = editor_window.findChild(QtWidgets.QDockWidget, "Script Canvas")
-    sc_main = sc.findChild(QtWidgets.QMainWindow)
-    create_new_graph = pyside_utils.find_child_by_pattern(
-        sc_main, {"objectName": "action_New_Script", "type": QtWidgets.QAction}
-    )
-    if sc.findChild(QtWidgets.QDockWidget, "NodeInspector") is None:
-        action = pyside_utils.find_child_by_pattern(sc, {"text": "Node Inspector", "type": QtWidgets.QAction})
-        action.trigger()
-    node_inspector = sc.findChild(QtWidgets.QDockWidget, "NodeInspector")
-    create_new_graph.trigger()
-
-    # 3) Add node
-    command_line_input("add_node Print")
-
-    # 4) Duplicate node
-    graph_view = sc.findChild(QtWidgets.QFrame, "graphicsViewFrame")
-    graph = graph_view.findChild(QtWidgets.QWidget, "")
-    # There are currently no utilities available to directly duplicate the node,
-    # therefore the node is selected using CTRL+A on the graph to select
-    # it and then CTRL+D to duplicate
-    sc_main.activateWindow()
-    QtTest.QTest.keyClick(graph, "a", Qt.ControlModifier, WAIT_FRAMES)
-    QtTest.QTest.keyClick(graph, "d", Qt.ControlModifier, WAIT_FRAMES)
-
-    # 5) Verify the node was duplicated
-    # As direct interaction with node is not available the text on the label
-    # inside the Node Inspector is validated showing two nodes exist
-    after_dup = grab_title_text()
-    Report.result(Tests.node_duplicated, after_dup == EXPECTED_STRING)
-
-
-if __name__ == "__main__":
-    import ImportPathHelper as imports
-
-    imports.init()
-    from editor_python_test_tools.utils import Report
-
-    Report.start_test(Node_HappyPath_DuplicateNode)
+    @pyside_utils.wrap_async
+    async def run_test(self):
+
+        # Preconditions
+        general.idle_enable(True)
+
+        # 1) Open Script Canvas window (Tools > Script Canvas)
+        general.open_pane(SCRIPT_CANVAS_UI)
+        helper.wait_for_condition(lambda: general.is_pane_visible(SCRIPT_CANVAS_UI), WAIT_TIME_3)
+
+        # 2) Open a new graph
+        editor_window = pyside_utils.get_editor_main_window()
+        sc_editor = editor_window.findChild(QtWidgets.QDockWidget, SCRIPT_CANVAS_UI)
+        sc_editor_main_window = sc_editor.findChild(QtWidgets.QMainWindow)
```

Comment:
This might be a good SC util definition to extract

```
def get_main_sc_editor_window(...)
   editor_window = pyside_utils.get_editor_main_window()
   sc_editor = editor_window.findChild(QtWidgets.QDockWidget, SCRIPT_CANVAS_UI)
   return sc_editor.findChild(QtWidgets.QMainWindow)
```

Afterwards we can write the following code to simplify the test script:

`sc_editor_main_window = get_main_sc_editor_window(...)`

Reply:
arg. tried doing this but the script is using the sc_editor object further down the line. I think this would be another script that would benefit from a qt model class since it and many other use similar 'get the main window and sc editor' behaviors.

Reply:
An alternative then is to move these to the preconditions or a local variables section or have them in an initialization class. Due to this being something we'd like to clean up into a qt model class, precons is probably the best route ATM.

A final solution would be the following.

```
def get_sc_editor_object(...)
   editor_window = pyside_utils.get_editor_main_window()
   return editor_window.findChild(QtWidgets.QDockWidget, SCRIPT_CANVAS_UI)

def get_main_sc_editor_window(...)
   sc_editor = get_sc_editor_object(...).findChild(QtWidgets.QDockWidget, SCRIPT_CANVAS_UI)
   return sc_editor.findChild(QtWidgets.QMainWindow)
```

This code pollutes the test steps ATM, and we should do our due diligence to do some cleanup at this time to make it clearer for us on a follow-up refactor where to go. So choosing any of the above solutions would be an improvement.

Reply:
Ok. I broke down the initialization behavior and moved it to the scripting tools file. It follows a similar format to what the other test scripts were doing so it should be pretty easy to refactor it once more for qt object model once we get there. will need to deprecate the previously added initialization functions though since they aren't specific enough to the SC areas such as node inspector, main window etc.